calendar/stream_implementer.bal

```diff
@@ -0,0 +1,169 @@
+    }
+
+    isolated function getAndConvertToEventArray(http:Response response) returns Event[]|error {
+        Event[] events = [];
+        map<json>|string? handledResponse = check handleResponse(response);
+        if (handledResponse is map<json>) {
+            self.nextLink = let var link = handledResponse["@odata.nextLink"]
+                in link is string ? link : EMPTY_STRING;
+            json values = check handledResponse.value;
+            if (values is json[]) {
+                foreach json item in values {
+                    Event convertedItem = check item.cloneWithType(Event);
+                    events.push(convertedItem);
+                }
+                return events;
+            } else {
+                return error(INVALID_PAYLOAD);
+            }
+        } else {
+            return error(INVALID_RESPONSE);
+        }
+    }
+}
+
+class CalendarStream {
+    private Calendar[] currentEntries = [];
+    private string nextLink;
+    private int index = 0;
+    private final http:Client httpClient;
+    private final string path;
+    private string? queryPraram;
+    Configuration config;
+
+    isolated function init(Configuration config, http:Client httpClient, string path, string? queryPraram = ()) 
+    returns error? {
+        self.httpClient = httpClient;
+        self.path = path;
+        self.queryPraram = queryPraram;
+        self.config = config;
+        self.nextLink = EMPTY_STRING;
+        self.currentEntries = check self.fetchRecordsInitial();
+    }
+
+    public isolated function next() returns record {|Calendar value;|}|error? {
+        if (self.index < self.currentEntries.length()) {
+            record {|Calendar value;|} singleRecord = {value: self.currentEntries[self.index]};
+            self.index += 1;
+            return singleRecord;
+        }
+        // This code block is for retrieving the next batch of records when the initial batch is finished.
+        if (self.nextLink != EMPTY_STRING && !self.queryPraram.toString().includes("$top")) {
+            self.index = 0;
+            self.currentEntries = check self.fetchRecordsNext();
+            record {|Calendar value;|} singleRecord = {value: self.currentEntries[self.index]};
+            self.index += 1;
+            return singleRecord;
+        }
+    }
+
+    isolated function fetchRecordsInitial() returns Calendar[]|error {
+        http:Response response 
+            = check self.httpClient->get(self.path);
+        _ = check handleResponse(response);
+        return check self.getAndConvertToCalendarArray(response);
+    }
+
+    isolated function fetchRecordsNext() returns Calendar[]|error {
+        http:Client nextPageClient = check new (self.nextLink, {
+            auth: self.config.clientConfig,
+            secureSocket: self.config?.secureSocketConfig
+        });
+        http:Response response 
+            = check nextPageClient->get(EMPTY_STRING);
```

Comment:
Shall we fix the formatting.
Please fix the same issue in other places also

Reply:
fixed in all places 
/Thanks